File: ScanbotSDKDemo/TextViewController.m
//
//  TextViewController.m
//  PayFormScannerDemo
//
//  Created by Sebastian Husche on 24.06.15.
//  Copyright (c) 2015 doo GmbH. All rights reserved.
//

#import "TextViewController.h"

@interface TextViewController ()
@property (strong, nonatomic) IBOutlet UITextView *textView;

@end

@implementation TextViewController

+ (TextViewController *)textControllerWithText:(NSString *)text {
    UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    TextViewController  *textViewController = (TextViewController *)[storyBoard instantiateViewControllerWithIdentifier:NSStringFromClass(self)];
    textViewController.textToDisplay = text;
    return textViewController;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self updateText];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:NO animated:animated];
}

- (void)setTextToDisplay:(NSString *)textToDisplay {
    _textToDisplay = [textToDisplay copy];
    [self updateText];
}

- (void)updateText {
    if (self.isViewLoaded) {
        self.textView.text = self.textToDisplay;
    }
}

@end

File: ScanbotSDKDemo/TextViewController.swift
//
//  TextViewController.swift
//  ScanbotSDKDemo
//
//  Created by Danil Voitenko on 25.11.21.
//  Copyright © 2021 doo GmbH. All rights reserved.
//

import UIKit

class TextViewController: UIViewController {
    @IBOutlet private var textView: UITextView?
    var textToDisplay: String? {
        didSet {
            updateText()
        }
    }
    
    static func make(with text: String) -> TextViewController {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        let controller = storyboard.instantiateViewController(withIdentifier: "TextViewController") as!
        TextViewController
        controller.textToDisplay = text
        
        return controller
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        updateText()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        navigationController?.setNavigationBarHidden(false, animated: animated)
    }
        
    private func updateText() {
        if isViewLoaded {
            textView?.text = textToDisplay
        }
    }
}
